src/udp/socket.rs

```diff
@@ -958,49 +963,117 @@ mod test {
 
     use util;
 
-    mod send_final_ack_acks {
-        use super::*;
-
-        fn localhost_addr(sock: &UdpSocket) -> SocketAddr {
-            let addr = unwrap!(sock.local_addr());
-            SocketAddr::new(ip!("127.0.0.1"), addr.port())
-        }
-
-        #[test]
-        fn it_sends_5_ack_ack_messages() {
-            let mut core = unwrap!(Core::new());
-            let handle = core.handle();
+    fn localhost_addr(sock: &UdpSocket) -> SocketAddr {
```

Comment:
Crust also has this sort of functionality https://github.com/maidsafe/crust/blob/dev/src/net/protocol_agnostic/addr.rs#L92
Maybe it would be worth implementing this in https://github.com/ustulation/p2p/blob/master/src/ip_addr.rs#L17 in the future?

Reply:
Definitely agree that this needs to go *somewhere*. Maybe a `net-utils` crate or something?

Reply:
Well, since `p2p` exports `IpAddrExt` (https://github.com/ustulation/p2p/blob/master/src/prelude.rs#L1) why not just put it there? :)